src/Servers/Kestrel/Core/test/HPackIntegerTests.cs

```diff
@@ -35,5 +37,55 @@ public void IntegerEncoderDecoderRoundtrips()
                 }
             }
         }
+
+        [Theory]
+        [MemberData(nameof(IntegerCodecSamples))]
+        public void EncodeSamples(int value, int bits, byte[] expectedResult)
+        {
+            Span<byte> actualResult = new byte[64];
```

Comment:
nit: I think this and `DecodeSamples` are a bit redundant with IntegerEncoderDecoderRoundtrips. I believe the combinations you listed would eventually be tested. the only difference is that you are verifying the expectedResult.